crates/oq3_semantics/src/asg.rs

```diff
@@ -562,6 +562,23 @@ impl GateDeclaration {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct MeasureExpression {
+    operand: Box<TExpr>,
+}
+
+impl MeasureExpression {
+    pub fn new(operand: TExpr) -> MeasureExpression {
+        MeasureExpression { operand: Box::new(operand) }
+    }
+
+    // FIXME: type may not be correct here.
+    // This assumes a single qubit is measured.
+    pub fn to_texpr(self) -> TExpr {
+        TExpr::new(Expr::MeasureExpression(self), Type::Bit(IsConst::False))
+    }
+}
```

Comment:
Don't forget a public getter for the `operand` so I can read it -
```rust
impl MeasureExpression {
    pub fn operand(&self) -> &TExpr {
        &self.operand
    }
}
```

Reply:
I hadn't forgotten. Just racing to get something that worked last night.